test/integration_tests/utils/model_output_result_number_checker.py

```diff
@@ -0,0 +1,28 @@
+def assert_correct_number_of_results(
+    added_columns: int,
+    removed_columns: int,
+    input_data_row: tuple,
+    result: list,
+    n_rows_result: int,
+):
+    n_cols_result = len(input_data_row) + (added_columns - removed_columns)
+    assert len(result) == n_rows_result and len(result[0]) == n_cols_result, (
```

Comment:
I would make a separate statement for checking the number of columns, like
`num_columns_ok = all(len(row) == n_cols_result for row in result)`
Then
`assert num_columns_ok and len(result) == n_rows_result`
Or, my personal preference is separate asserts - easier to see where it breaks and you can make the error messages more specific.

Reply:
torsten often prefers one assert statement. i am fine with both versions, i dont think it makes that much of a difference here